x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/inference/trainedmodel/NlpConfigUpdate.java

```diff
@@ -7,6 +7,80 @@
 
 package org.elasticsearch.xpack.core.ml.inference.trainedmodel;
 
-public abstract class NlpConfigUpdate implements InferenceConfigUpdate {
+import org.elasticsearch.Version;
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.core.Nullable;
+import org.elasticsearch.xcontent.ToXContent;
+import org.elasticsearch.xcontent.XContentBuilder;
+import org.elasticsearch.xpack.core.ml.utils.ExceptionsHelper;
+import org.elasticsearch.xpack.core.ml.utils.NamedXContentObject;
+import org.elasticsearch.xpack.core.ml.utils.NamedXContentObjectHelper;
 
+import java.io.IOException;
+import java.util.Map;
+
+public abstract class NlpConfigUpdate implements InferenceConfigUpdate, NamedXContentObject {
+
+    @SuppressWarnings("unchecked")
+    public static TokenizationUpdate tokenizationFromMap(Map<String, Object> map) {
+        Map<String, Object> tokenziation = (Map<String, Object>) map.remove("tokenization");
+        if (tokenziation == null) {
+            return null;
+        }
+
+        Map<String, Object> bert = (Map<String, Object>) tokenziation.remove("bert");
+        if (bert == null && tokenziation.isEmpty() == false) {
+            throw ExceptionsHelper.badRequestException("unknown tokenization type expecting one of [bert] got {}", tokenziation.keySet());
+        }
+        Object truncate = bert.remove("truncate");
+        return new BertTokenizationUpdate(Tokenization.Truncate.fromString(truncate.toString()));
```

Comment:
Its possible that `bert: {}` is provided, so `truncate` would be `null` and then throw an NPE when you call `.toString()`

Reply:
👍 I added a test